src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -152,46 +138,49 @@ public string GenerateStartTag()
         /// <summary>
         /// Generates the <see cref="TagHelperOutput"/>'s <see cref="PreContent"/>.
         /// </summary>
-        /// <returns><c>string.Empty</c> if <see cref="TagName"/> is not <c>null</c> or whitespace
-        /// and <see cref="SelfClosing"/> is <c>true</c>. Otherwise, <see cref="PreContent"/>.</returns>
-        public string GeneratePreContent()
+        /// <returns><c>null</c> if <see cref="TagName"/> is not <c>null</c> or whitespace
+        /// and <see cref="SelfClosing"/> is <c>true</c>.
+        /// Otherwise, returns <see cref="ITextWriterCopyable"/>.</returns>
```

Comment:
`Otherwise, an <see cref="ITextWriterCopyable"/> containing the <see cref="PreContent"/> content.`; same for other `Generates*()` descriptions